private_sharing/forms.py

```diff
@@ -97,6 +113,44 @@ def clean(self):
                                                      'member accounts.'))
         return cleaned_data
 
+    def save(self, **kwargs):
+        """
+        Override save to save requested sources in approrpiate table.
```

Comment:
typo

Reply:
and this is a lot of new & custom looking code -- and I wonder if this might all be more straightforward if you use the manytomanyfield along with the "through"